dashboard/app/models/user_level.rb

```diff
@@ -94,7 +95,28 @@ def paired?
     driver? || navigator?
   end
 
-  def handle_unsubmit
+  def submitted_or_resubmitted?
+    submitted_changed?(to: true) || (submitted? && level_source_id_changed?)
+  end
```

Comment:
I don't love the additional complexity here but I believe this sticks closer to the intended behavior: If this UserLevel just _became_ `submitted` -or- it was already `submitted` and the user's answer (`level_source_id`) changed, then this save represents a (re?)submission.

Note: I discovered that we can't just fall back on _any_ save when `submitted` is true, because of this hook in `PeerReview`:

https://github.com/code-dot-org/code-dot-org/blob/50b867a0c44a1c90a6e39c3b33b482e992e218d8/dashboard/app/models/peer_review.rb#L41

So we get this semi-circular dependency where updating a `PeerReview` could update its `UserLevel` which could in turn create more `PeerReviews`.  I think that's a good argument for being fairly precise in this update condition.